### PR TITLE
Select component

### PR DESCRIPTION
**Pull request by turquoisemelon**
### Screenshot
Screen Shot 2020-03-20 at 2 46 51 PM

### Designs

https://www.figma.com/file/RqENxZWAzGiEWM7COch1Sc/Radius-Design-Kit?node-id=542%3A6317

### CODE REVIEW

File: packages/ds/src/components/select/select.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      variant="body"
+      ref={ref}
+      py={2}
+      hasLeftIcon={!!leftIcon}
+      hasRightIcon={!!rightIcon}
+      disabled={disabled}
+      aria-invalid={props.error ? 'true' : 'false'}
+      {...props}
+    />
+    {rightIcon && (
```
Comment (winkerVSbecks) [inline packages/ds/src/components/select/select.tsx]: The right icon would just always be the `ExpandMore`, right?

Comment (winkerVSbecks) [reply packages/ds/src/components/select/select.tsx]: sorry, never mind. Just saw the figma specs

Comment (turquoisemelon) [reply packages/ds/src/components/select/select.tsx]: We have two right icons actually. One is `ExpandMore` which doesn't change and second one sits left side of `ExpandMore` and is optional. Designs are here:
https://www.figma.com/file/RqENxZWAzGiEWM7COch1Sc/Radius-Design-Kit?node-id=542%3A6317

I thought `rightIcon` styled-component could be leveraged for both so kept its name. Though, open to suggestions as renaming or restructuring it

**Comment by winkerVSbecks** [comment]
@turquoisemelon you can probably set `pointer-events: none` on the ExpandMore icon.

File: packages/ds/src/components/select/select.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      >
+        <option>Pina Colada</option>
+        <option>Margarita</option>
+        <option>Mojito</option>
+      </Select>
+      <Select
+        error
+        defaultValue="Margarita"
+        leftIcon={<Icons.Search color="text.primary" aria-hidden />}
+        rightIcon={<Icons.Info color="text.primary" aria-hidden />}
```
Comment (winkerVSbecks) [inline packages/ds/src/components/select/select.stories.mdx]: Should be `<Icons.Error aria-hidden color="ui.error" />` for all the error stories

**Comment by winkerVSbecks** [comment]
@turquoisemelon I would also recommend adding Select story to `form-control/field.stories`

**Comment by winkerVSbecks** [comment]
A few of the stories need to be updated to account for bg name change. Check out chromatic for breaking changes.

**Comment by turquoisemelon** [comment]
> A few of the stories need to be updated to account for bg name change. Check out chromatic for breaking changes. And the theme decorator.

Sure, will do 👍 One last thing I need to do is kitchen sinking the component to sample app. Should be good to go once I do those.

**Comment by winkerVSbecks** [comment]
Also this can totally wait till Monday. No rush.

File: packages/ds/src/components/select/select.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        mb={4}
+      >
+        <option>Select a city</option>
+        <option>Toronto</option>
+        <option>Istanbul</option>
+        <option>Amsterdam</option>
+      </Select>
+      <Select
+        defaultValue="Select a city"
+        leftIcon={<Icons.Place color="text.primary" aria-hidden />}
```
Comment (winkerVSbecks) [inline packages/ds/src/components/select/select.stories.mdx]: The icon color is not working as expected. In has the wrong color in dark mode.

Comment (turquoisemelon) [reply packages/ds/src/components/select/select.stories.mdx]: Thanks for the catch, looking into it 👍

**Pull request merged**